packages/chain-helpers/src/blockchain/Blockchain.ts

```diff
@@ -167,26 +173,71 @@ export async function submitSignedTx(
 
 export const dispatchTx = submitSignedTx
 
+const metadataHashes = new Map<string, HexString>()
+
+// Returns the Merkle root of the metadata as stored in the `ConfigService` cache. If not present, it computes it, stores it in the cache for future retrievals, and returns it.
+async function getMetadataHash(api: ApiPromise): Promise<HexString> {
+  const metadata = await api.call.metadata.metadataAtVersion(15)
+  // TODO: Find out why using this metadata here fails to decode when calculating the Merkle root.
+  // const metadata = api.runtimeMetadata.asV15
+  const { specName, specVersion } = api.runtimeVersion
+  const genesisHash = await api.genesisHash
+  const cacheKey = blake2AsHex(
+    Uint8Array.from([
+      ...specName.toU8a(),
+      ...specVersion.toU8a(),
+      ...genesisHash.toU8a(),
+    ])
+  )
+  if (metadataHashes.has(cacheKey)) {
+    return metadataHashes.get(cacheKey) as HexString
+  }
+  const merkleInfo: ExtraInfo = {
+    base58Prefix: api.consts.system.ss58Prefix.toNumber(),
+    decimals: api.registry.chainDecimals[0],
+    specName: specName.toString(),
+    specVersion: specVersion.toNumber(),
+    tokenSymbol: api.registry.chainTokens[0],
+  }
+  const merkleizedMetadata = merkleizeMetadata(metadata.toHex(), merkleInfo)
```

Comment:
This seems to be the earliest point where the metadata is needed

Reply:
Yes sorry I left the PR with one more open point I did not remember about. I changed the fetching here: https://github.com/KILTprotocol/sdk-js/pull/924/commits/4b05d7251ee8fc96bc36296cca0b92d20f75c95e.

There is still an error if we try to use the locally retrieved metadata, and I believe there should be one transformation step somehow involved, but I think for now we could go ahead and merge this, and we can optimise this in the future. Also, the fetching happens once and it's then cached, so not too big of a deal.

Reply:
> I think for now we could go ahead and merge this, and we can optimise this in the future. Also, the fetching happens once and it's then cached, so not too big of a deal.

Yes, my thoughts exactly